ement-room.el

```diff
@@ -2787,135 +2800,118 @@ ROOM defaults to the value of `ement-room'."
 
 ;; ;; ;; URLs
 
-(defvar-local ement-room--preview-stop nil
-  "Whether to proceed to previews for current room or not. Used in case of rate limit.")
-
-(defun ement-room--linkify-urls (string msg)
+(defun ement-room--linkify-urls (string message)
   "Return STRING with URLs in it made clickable.
-MSG is the message STRING was contained in."
+MESSAGE is the message STRING was contained in."
   ;; Is there an existing Emacs function to do this?  I couldn't find one.
   ;; Yes, maybe: `goto-address-mode'.  TODO: Try goto-address-mode.
   (let ((buffer (alist-get 'buffer (ement-room-local ement-room))))
     (with-temp-buffer
       (insert string)
       (goto-char (point-min))
       (cl-loop with c = 1
+               with url = nil
                while (re-search-forward (rx bow "http" (optional "s") "://" (1+ (not space)))
                                         nil 'noerror)
-               do (make-text-button (match-beginning 0) (match-end 0)
-                                    'mouse-face 'highlight
-                                    'face 'link
-                                    'help-echo (match-string 0)
-                                    'action #'browse-url-at-mouse
-                                    'follow-link t)
+               do (setf url (buffer-substring-no-properties (match-beginning 0) (match-end 0)))
+               (make-text-button (match-beginning 0) (match-end 0)
+                                 'mouse-face 'highlight
+                                 'face 'link
+                                 'help-echo url
+                                 'action #'browse-url-at-mouse
+                                 'follow-link t)
                if (and ement-room-url-previews
-                       (not (map-elt (map-elt (ement-event-content msg) 'm.relates_to)
-                                     'm.in_reply_to))
+                       (not (map-nested-elt (ement-event-content message) '(m.relates_to m.in_reply_to)))
                        ;; spec recommends not getting previews for encrypted rooms
                        (not (with-current-buffer buffer
                               (cl-find-if (apply-partially #'equal "m.room.encryption")
                                           (ement-room-invite-state ement-room)
                                           :key #'ement-event-type))))
-               do (let ((string (buffer-substring-no-properties (match-beginning 0) (match-end 0))))
-                    (with-current-buffer buffer
-                      (ement-room--get-preview msg string c)))
+               do (with-current-buffer buffer
+                    (ement-room--url-preview message url c))
                (setf c (1+ c)))
       (buffer-string))))
 
-(defun ement-room--get-preview (msg link num)
-  "Get preview of LINK. MSG is the message this link is from. NUM is the index for this link."
-  (let* ((id (ement-event-id msg))
+(defun ement-room--url-preview (message link num)
```

Comment:
Please follow the conventions regarding argument names in the project.  For example, here `message` is actually an `ement-event` struct, so the argument should be named `event`.